libskale/State.cpp

```diff
@@ -348,7 +355,7 @@ void State::commit( CommitBehaviour _commitBehaviour ) {
             }
         }
         m_db_ptr->updateStorageUsage( totalStorageUsed_ );
-        m_db_ptr->commit();
+        m_db_ptr->commit( fn_pre_commit );
```

Comment:
Why can't we pass last transaction hash to OverlayDB in the same way as we pass totalStorageUsed_?
Do we really need this callback mechanism?

Reply:
We cannot pass last transaction hash. This is not what is needed here. We need to do operations which modify batch actions created in memory. This is same as transactions in classic SQL. Function handler to add level DB batch actions looks like the only way

Reply:
Yes, of course we need it in batch. But what's the problem to have "last transaction hash" field in OverlayDB class and have it put this field writing to DB into batch?